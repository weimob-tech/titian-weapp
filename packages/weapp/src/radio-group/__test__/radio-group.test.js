const path = require('path');
const simulate = require('miniprogram-simulate');

const { getSystemInfoSync } = wx;

global.wx.getSystemInfoSync = () => {
  const res = getSystemInfoSync();
  return {
    ...res,
    SDKVersion: '2.9.3'
  };
};

describe('test component', () => {
  const id = simulate.load(path.resolve(__dirname, './index'), {
    rootPath: path.join(process.cwd(), 'packages/weapp', 'src')
  });

  it('should render success', async () => {
    const container = simulate.render(id);
    container.attach(document.createElement('parent-wrapper'));
    await simulate.sleep(40);
    expect(container.toJSON()).toMatchSnapshot();

    const radioIconWrap = container.querySelector('.radio >>> .titian-radio-icon-wrap');
    const radio2IconWrap = container.querySelector('.radio2 >>> .titian-radio-icon-wrap');

    radioIconWrap.dispatchEvent('tap');
    await simulate.sleep(80);
    expect(container.data.selectValue).toBe('1');
    expect(container.toJSON()).toMatchSnapshot();

    radio2IconWrap.dispatchEvent('tap');
    await simulate.sleep(80);
    expect(container.data.selectValue).toBe('2');
    expect(container.toJSON()).toMatchSnapshot();
  });
});
