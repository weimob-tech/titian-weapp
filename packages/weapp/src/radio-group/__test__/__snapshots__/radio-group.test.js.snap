// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component should render success 1`] = `
<main>
  <ti-radio-group
    name="select"
    bind:change="onChange"
  >
    <wx-view
      class="ext-class titian-radio-group"
      style=""
    >
      <ti-radio
        class="radio"
      >
        <wx-view
          class="ext-class titian-radio"
        >
          <wx-view
            class="titian-radio-icon-wrap"
            style="height:32rpx;--base-radius-size:32rpx"
            bind:tap="onChange"
          >
            <ti-icon
              extClass="icon-class titian-radio-icon-animation titian-radio-icon"
            >
              <wx-view
                class="titian-icon-base titian-icon titian-icon-checkbox-hollow ext-class"
                style="line-height:32rpx;height:32rpx;width:32rpx;font-size:32rpx"
                bind:tap="onClick"
              />
            </ti-icon>
          </wx-view>
          <wx-view
            class="titian-radio-label"
            bind:tap="onClick"
          >
            选项1
          </wx-view>
        </wx-view>
      </ti-radio>
      <ti-radio
        class="radio2"
      >
        <wx-view
          class="ext-class titian-radio"
        >
          <wx-view
            class="titian-radio-icon-wrap"
            style="height:32rpx;--base-radius-size:32rpx"
            bind:tap="onChange"
          >
            <ti-icon
              extClass="icon-class titian-radio-icon-animation titian-radio-icon"
            >
              <wx-view
                class="titian-icon-base titian-icon titian-icon-checkbox-hollow ext-class"
                style="line-height:32rpx;height:32rpx;width:32rpx;font-size:32rpx"
                bind:tap="onClick"
              />
            </ti-icon>
          </wx-view>
          <wx-view
            class="titian-radio-label"
            bind:tap="onClick"
          >
            选项2
          </wx-view>
        </wx-view>
      </ti-radio>
      <ti-radio>
        <wx-view
          class="ext-class titian-radio-checked titian-radio"
        >
          <wx-view
            class="titian-radio-icon-wrap-checked titian-radio-icon-wrap"
            style="height:32rpx;--base-radius-size:32rpx"
            bind:tap="onChange"
          >
            <ti-icon
              extClass="icon-class titian-radio-icon-zoom-in titian-radio-icon-checked titian-radio-icon-animation titian-radio-icon"
            >
              <wx-view
                class="titian-icon-base titian-icon titian-icon-checkbox-hollow ext-class"
                style="line-height:32rpx;height:32rpx;width:32rpx;font-size:32rpx"
                bind:tap="onClick"
              />
            </ti-icon>
          </wx-view>
          <wx-view
            class="titian-radio-label"
            bind:tap="onClick"
          >
            选项3
          </wx-view>
        </wx-view>
      </ti-radio>
    </wx-view>
  </ti-radio-group>
</main>
`;

exports[`test component should render success 2`] = `
<main>
  <ti-radio-group
    name="select"
    bind:change="onChange"
  >
    <wx-view
      class="ext-class titian-radio-group"
      style=""
    >
      <ti-radio
        class="radio"
      >
        <wx-view
          class="ext-class titian-radio-checked titian-radio"
        >
          <wx-view
            class="titian-radio-icon-wrap-checked titian-radio-icon-wrap"
            style="height:32rpx;--base-radius-size:32rpx"
            bind:tap="onChange"
          >
            <ti-icon
              extClass="icon-class titian-radio-icon-zoom-in titian-radio-icon-checked titian-radio-icon-animation titian-radio-icon"
            >
              <wx-view
                class="titian-icon-base titian-icon titian-icon-checkbox-hollow ext-class"
                style="line-height:32rpx;height:32rpx;width:32rpx;font-size:32rpx"
                bind:tap="onClick"
              />
            </ti-icon>
          </wx-view>
          <wx-view
            class="titian-radio-label"
            bind:tap="onClick"
          >
            选项1
          </wx-view>
        </wx-view>
      </ti-radio>
      <ti-radio
        class="radio2"
      >
        <wx-view
          class="ext-class titian-radio"
        >
          <wx-view
            class="titian-radio-icon-wrap"
            style="height:32rpx;--base-radius-size:32rpx"
            bind:tap="onChange"
          >
            <ti-icon
              extClass="icon-class titian-radio-icon-animation titian-radio-icon"
            >
              <wx-view
                class="titian-icon-base titian-icon titian-icon-checkbox-hollow ext-class"
                style="line-height:32rpx;height:32rpx;width:32rpx;font-size:32rpx"
                bind:tap="onClick"
              />
            </ti-icon>
          </wx-view>
          <wx-view
            class="titian-radio-label"
            bind:tap="onClick"
          >
            选项2
          </wx-view>
        </wx-view>
      </ti-radio>
      <ti-radio>
        <wx-view
          class="ext-class titian-radio"
        >
          <wx-view
            class="titian-radio-icon-wrap"
            style="height:32rpx;--base-radius-size:32rpx"
            bind:tap="onChange"
          >
            <ti-icon
              extClass="icon-class titian-radio-icon-animation titian-radio-icon"
            >
              <wx-view
                class="titian-icon-base titian-icon titian-icon-checkbox-hollow ext-class"
                style="line-height:32rpx;height:32rpx;width:32rpx;font-size:32rpx"
                bind:tap="onClick"
              />
            </ti-icon>
          </wx-view>
          <wx-view
            class="titian-radio-label"
            bind:tap="onClick"
          >
            选项3
          </wx-view>
        </wx-view>
      </ti-radio>
    </wx-view>
  </ti-radio-group>
</main>
`;

exports[`test component should render success 3`] = `
<main>
  <ti-radio-group
    name="select"
    bind:change="onChange"
  >
    <wx-view
      class="ext-class titian-radio-group"
      style=""
    >
      <ti-radio
        class="radio"
      >
        <wx-view
          class="ext-class titian-radio"
        >
          <wx-view
            class="titian-radio-icon-wrap"
            style="height:32rpx;--base-radius-size:32rpx"
            bind:tap="onChange"
          >
            <ti-icon
              extClass="icon-class titian-radio-icon-zoom-out titian-radio-icon-animation titian-radio-icon"
            >
              <wx-view
                class="titian-icon-base titian-icon titian-icon-checkbox-hollow ext-class"
                style="line-height:32rpx;height:32rpx;width:32rpx;font-size:32rpx"
                bind:tap="onClick"
              />
            </ti-icon>
          </wx-view>
          <wx-view
            class="titian-radio-label"
            bind:tap="onClick"
          >
            选项1
          </wx-view>
        </wx-view>
      </ti-radio>
      <ti-radio
        class="radio2"
      >
        <wx-view
          class="ext-class titian-radio-checked titian-radio"
        >
          <wx-view
            class="titian-radio-icon-wrap-checked titian-radio-icon-wrap"
            style="height:32rpx;--base-radius-size:32rpx"
            bind:tap="onChange"
          >
            <ti-icon
              extClass="icon-class titian-radio-icon-zoom-in titian-radio-icon-checked titian-radio-icon-animation titian-radio-icon"
            >
              <wx-view
                class="titian-icon-base titian-icon titian-icon-checkbox-hollow ext-class"
                style="line-height:32rpx;height:32rpx;width:32rpx;font-size:32rpx"
                bind:tap="onClick"
              />
            </ti-icon>
          </wx-view>
          <wx-view
            class="titian-radio-label"
            bind:tap="onClick"
          >
            选项2
          </wx-view>
        </wx-view>
      </ti-radio>
      <ti-radio>
        <wx-view
          class="ext-class titian-radio"
        >
          <wx-view
            class="titian-radio-icon-wrap"
            style="height:32rpx;--base-radius-size:32rpx"
            bind:tap="onChange"
          >
            <ti-icon
              extClass="icon-class titian-radio-icon-animation titian-radio-icon"
            >
              <wx-view
                class="titian-icon-base titian-icon titian-icon-checkbox-hollow ext-class"
                style="line-height:32rpx;height:32rpx;width:32rpx;font-size:32rpx"
                bind:tap="onClick"
              />
            </ti-icon>
          </wx-view>
          <wx-view
            class="titian-radio-label"
            bind:tap="onClick"
          >
            选项3
          </wx-view>
        </wx-view>
      </ti-radio>
    </wx-view>
  </ti-radio-group>
</main>
`;
